src/test/java/org/assertj/core/error/ShouldHaveCauseExactlyInstance_create_Test.java

```diff
@@ -32,4 +35,21 @@ void should_create_error_message_for_no_cause() {
       "  %s%n" +
       "but current throwable has no cause.", expected));
   }
+
+  @Test
+  void should_create_error_message_for_wrong_cause() {
+    // GIVEN
+    Throwable expected = new IllegalStateException();
+    Throwable cause = new IllegalAccessError();
+    Throwable actual = new RuntimeException(cause);
+    // WHEN
+    String message = shouldHaveCauseInstance(actual, expected.getClass()).create();
+    // THEN
+    then(message).isEqualTo(format("%nExpecting a throwable with cause being an instance of:%n" +
+      "  %s%n" +
+      "but was an instance of:%n" +
+      "  %s%n" +
+      "Throwable that failed the check:%n" +
+      "%n" + escapePercent(getStackTrace(actual)), expected, cause));
+  }
```

Comment:
`isEqualTo` for String as an overload version that supports the `String.format` syntax, you can then remove the call to `format` here.

Also can you add a cause message containing a `%` to make sure it does not break the code?

see https://www.javadoc.io/doc/org.assertj/assertj-core/latest/org/assertj/core/api/AbstractStringAssert.html

this comment applies to the other tests too ;-)